worlds/jakanddaxter/__init__.py

```diff
@@ -0,0 +1,493 @@
+        elif item in range(jak1_id + Orbs.orb_offset, jak1_max - max(trap_item_table)):
+            counts_and_classes.append((self.total_prog_orb_bundles, ItemClass.progression_skip_balancing))
+            counts_and_classes.append((self.total_filler_orb_bundles, ItemClass.filler))
+
+        # We will manually create trap items as needed.
+        elif item in range(jak1_max - max(trap_item_table), jak1_max):
+            counts_and_classes.append((0, ItemClass.trap))
+
+        # We will manually create filler items as needed.
+        elif item == jak1_max:
+            counts_and_classes.append((0, ItemClass.filler))
+
+        # If we try to make items with ID's higher than we've defined, something has gone wrong.
+        else:
+            raise KeyError(f"Tried to fill item pool with unknown ID {item}.")
+
+        return counts_and_classes
+
+    def create_items(self) -> None:
+        items_made: int = 0
+        for item_name in self.item_name_to_id:
+            item_id = self.item_name_to_id[item_name]
+
+            # Handle Move Randomizer option.
+            # If it is OFF, put all moves in your starting inventory instead of the item pool,
+            # then fill the item pool with a corresponding amount of filler items.
+            if item_name in self.item_name_groups["Moves"] and not self.options.enable_move_randomizer:
+                self.multiworld.push_precollected(self.create_item(item_name))
+                self.multiworld.itempool.append(self.create_filler())
+                items_made += 1
+                continue
+
+            # Handle Orbsanity option.
+            # If it is OFF, don't add any orb bundles to the item pool, period.
+            # If it is ON, don't add any orb bundles that don't match the chosen option.
+            if (item_name in self.item_name_groups["Precursor Orbs"]
+                    and (self.options.enable_orbsanity == Options.EnableOrbsanity.option_off
+                         or item_name != self.orb_bundle_item_name)):
+                continue
+
+            # Skip Traps for now.
+            if item_name in self.item_name_groups["Traps"]:
+                continue
+
+            # In almost every other scenario, do this. Not all items with the same name will have the same item class.
+            counts_and_classes = self.item_type_helper(item_id)
+            for (count, classification) in counts_and_classes:
+                self.multiworld.itempool += [JakAndDaxterItem(item_name, classification, item_id, self.player)
+                                             for _ in range(count)]
+                items_made += count
+
+        # Handle Traps (for real).
+        # Manually fill the item pool with a weighted assortment of trap items, equal to the sum of
+        # total_trap_cells + total_trap_orb_bundles. Only do this if one or more traps have weights > 0.
+        names, weights = self.trap_weights
+        if sum(weights) > 0:
```

Comment:
Can also drop the > here unless this can be negative somehow.

Reply:
If this could be negative, `random.choices` would throw a VlaueError and then the player's got bigger problems.